lib/LLVMIRCodeGen/Pipeline.cpp

```diff
@@ -63,6 +63,42 @@ bool LLVMIRGen::preserveSymbol(const llvm::GlobalValue &GV) {
   return true;
 }
 
+llvm::Attribute::AttrKind
+LLVMIRGen::getInlinineAttr(const llvm::Function *F) const {
+  return llvm::Attribute::AttrKind::None;
+}
+
+void LLVMIRGen::updateInlineAttributes(llvm::Module *M) {
+  // An empty attribute set to replace the target-specific machine code
+  // attributes that were attached by the frontend.
+  llvm::AttributeList AL;
+  for (auto &FF : *M) {
+    if (FF.isDeclaration()) {
+      continue;
+    }
+    // Check for no-inline attribute.
+    bool dontInline = FF.hasFnAttribute(llvm::Attribute::AttrKind::NoInline);
+    bool alwaysInline =
+        FF.hasFnAttribute(llvm::Attribute::AttrKind::AlwaysInline);
+    auto inlineAttr = getInlinineAttr(&FF);
+    if (inlineAttr != llvm::Attribute::AttrKind::None) {
+      DCHECK(inlineAttr == llvm::Attribute::AttrKind::AlwaysInline ||
+             inlineAttr == llvm::Attribute::AttrKind::NoInline)
+          << "Unknown inlining attribute returned by getInlinineAttr";
+      dontInline = (inlineAttr == llvm::Attribute::AttrKind::NoInline);
+    }
+    // Clear all attributes.
+    FF.setAttributes(AL);
```

Comment:
I am not sure about this: we are clearing *all* attributes, not just inline, and not restoring the others for now.
This is fine, for now, but what If someone adds another llvm attribute in the future? could be a recipe for future bugs..

Reply:
Good catch, this however seems to be the old behaviour (this code was moved).

Reply:
@shajrawi Sameer is right. We had this logic since ever. But I agree that it is currently not very easy to compose with other attributes.
I just tried to limit the scope and to remove just inlining related attributes, i.e. NoInline and InlineAlways. That in turn required removing OptimizeNone attribute as well, as it cannot exist without NoInline. Once I've done that, some of the CPU tests started failing because they now produce wrong results. I'm not quite sure why this happens, but it seems like it is better to leave the old logic in place for now.